### DISCUSSION TITLE
Convert Vanara.PInvoke.FirewallApi from VB.NET to C#

### DISCUSSION THREAD

**Discussion by NN---** [dahall/Vanara#216] [General]
Is there any reason why FirewallApi is VB.NET and not C# ?
https://github.com/dahall/Vanara/tree/master/PInvoke/FirewallApi

Can it be converted ?

**Comment by dahall** [comment]
In the release notes for 3.2.18, I put the following:
"Initial implementation of FirewallApi.dll. Yes, it is in VB instead of C#. Only VB supports parameterized properties and the underlying COM objects have these."

It can be converted, but then those nice parameterized properties have to be converted to methods.

**Reply by NN---** [reply]
Oh. I see. Either way C# doesn’t have it so using methods gets the same result.
The advantage is that C# has better code analysis than VB.
